### DISCUSSION TITLE
When there is only one result, it redirects to that result instead of just listing it in the list of results

### DISCUSSION THREAD

**Discussion by gagliardetto** [community/community#8622] [Other Feature Feedback, Questions, & Ideas] [answered]
Scenario: for a given query, there is only one result found.

What happens: after the search successfully completes, you get redirected to that result.

Expected behaviour: you see the one result in the result list just like you would when there are more than one results.

**Comment by look** [comment]
Thanks for the feedback @gagliardetto.

Can you explain more about why you want to see the result listing rather than going straight to the file? Thanks!

**Reply by gagliardetto** [reply]
The redirect to the result is an unexpected outcome.

I have no way of knowing that the query has only one result and think that maybe I clicked somewhere by accident.

Also, that redirection to the result page switches the search scope from `all` to only the repo/user of that first result.

**Reply by gagliardetto** [reply]
My reason is this: Just because there is only one result, it doesn't mean I want to open it.

**Reply by Savageman** [reply]
I was very surprised by this behaviour as well! I did think of "maybe this is the only result", but had no way to check whether it was actually the case or not

**Reply by gagliardetto** [reply]
Yeah exactly. Made me think maybe I accidentally clicked something.

**Reply by ste93cry** [reply]
>My reason is this: Just because there is only one result, it doesn't mean I want to open it.

I agree with this. For me, the biggest downside was that I was more interested in the full path of the file rather than the content and while the results list clearly states the repository where the file belongs to, the content view does not

**Comment by colinwm** [comment, marked as answer]
OK, we've removed this behaviour. Now if you have one result, you'll still get the search results page.

**Reply by gagliardetto** [reply]
Thanks!
